### PR TITLE
Detect account-contracts

### PR DESCRIPTION
**Pull request by kasperski95**
Protostar will try to again compiling the contract if the account contract error is detected.

### CODE REVIEW

File: protostar/utils/starknet_compilation.py [hunk 1]
```diff
@@ -67,37 +67,35 @@ def preprocess_contract(
...
             )
-            assert isinstance(assembled, ContractClass)
-            return assembled
         except PreprocessorError as err:
-            err_message = err.message
-            if isinstance(err_message, str) and "account_contract" in err_message:
-                raise ProtostarException(
-                    err_message.replace("account_contract", "account-contract")
-                ) from err
+            if isinstance(err.message, str) and "account_contract" in err.message:
```
Comment (MaksymilianDemitraszek) [inline protostar/utils/starknet_compilation.py:81]: Can we add info about it to documentation? I think it's worth mentioning

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]

File: protostar/utils/starknet_compilation.py (hunk 1 above)
Comment (kasperski95) [reply protostar/utils/starknet_compilation.py:81]: Should user care about that? What's should be in the docs?

Comment (MaksymilianDemitraszek) [reply protostar/utils/starknet_compilation.py:81]: That we automatically detect that, users can come from starknet-compiler which requires this flag.

Comment (MaksymilianDemitraszek) [reply protostar/utils/starknet_compilation.py:81]: I know, but it exists, please provide it as a ::tip or something

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Pull request merged**